roxie/ccd/ccdsnmp.cpp

```diff
@@ -997,17 +916,13 @@ class CQueryStatsAggregator : public CInterface, implements IQueryStatsAggregato
         if (expirySeconds && difftime(timeNow, from) <= expirySeconds)
         {
             // we can calculate exactly
-            CIArrayOf<QueryStatsRecord> useStats;
+            std::deque<QueryStatsRecord> useStats;
             {
                 SpinBlock b(lock); // be careful not to take too much time in here! If it gets to take a while, we will have to rethink
-                ForEachQueueItemIn(idx, recent)
+                for (auto rec : recent)
                 {
-                    QueryStatsRecord *rec = recent.item(idx);
-                    if (rec->inRange(from, to))
-                    {
-                        rec->Link();
-                        useStats.append(*rec);
-                    }
+                    if (rec.inRange(from, to))
+                        useStats.push_back(rec);
```

Comment:
This will now potentially take a bit longer since it will be cloning a record - but will be removing an atomic operation, so probably fairly comparable.

Reply:
Modern compilers clone small records very efficiently.